common/src/main/java/dev/architectury/event/events/common/InteractionEvent.java

```diff
@@ -0,0 +1,160 @@
+    }
+    
+    interface LeftClickBlock {
+        /**
+         * Invoked whenever a player left clicks a block.
+         * Equivalent to Forge's {@code PlayerInteractEvent.LeftClickBlock} event and Fabric's {@code AttackBlockCallback}.
+         *
+         * @param player The player left clicking the block.
+         * @param hand   The hand that is used.
+         * @param pos    The position of the block in the level. Use {@link Player#getCommandSenderWorld()} to get the level.
+         * @param face   The face of the block clicked.
+         * @return A {@link EventResult} determining the outcome of the event,
+         * the execution of the vanilla click may be cancelled by the result.
+         */
+        EventResult click(Player player, InteractionHand hand, BlockPos pos, Direction face);
+    }
+    
+    interface RightClickItem {
+        /**
+         * Invoked whenever a player uses an item on a block.
+         * Equivalent to Forge's {@code PlayerInteractEvent.RightClickItem} event and Fabric's {@code UseItemCallback}.
+         *
+         * @param player The player right clicking the block.
+         * @param hand   The hand that is used.
+         * @return A {@link EventResult} determining the outcome of the event,
+         * the execution of the vanilla click may be cancelled by the result.
+         */
+        CompoundEventResult<ItemStack> click(Player player, InteractionHand hand);
+    }
+    
+    interface ClientRightClickAir {
+        /**
+         * Invoked whenever a player right clicks the air.
+         * This only occurs on the client.
+         * Equivalent to Forge's {@code PlayerInteractEvent.RightClickEmpty} event.
+         *
+         * @param player The player. Always {@link net.minecraft.client.player.LocalPlayer}
+         * @param hand   The hand used.
+         */
+        void click(Player player, InteractionHand hand);
+    }
+    
+    interface ClientLeftClickAir {
+        /**
+         * Invoked whenever a player left clicks the air.
+         * This only occurs on the client.
+         * Equivalent to Forge's {@code PlayerInteractEvent.LeftClickEmpty} event.
+         *
+         * @param player The player. Always {@link net.minecraft.client.player.LocalPlayer}
+         * @param hand   The hand used.
+         */
+        void click(Player player, InteractionHand hand);
+    }
+    
+    interface InteractEntity {
+        /**
+         * Invoked whenever a player right clicks an entity.
+         * Equivalent to Forge's {@code PlayerInteractEvent.EntityInteract} event.
+         *
+         * @param player The player clicking the entity.
+         * @param entity Then entity the player clicks.
+         * @param hand   The used hand.
+         * @return A {@link EventResult} determining the outcome of the event,
+         * the execution of the vanilla interaction may be cancelled by the result.
```

Comment:
Could be shortened to just
```suggestion
         * the action may be cancelled by the result.
```